### PR TITLE
all: fix uses of errors.As()

### PR DESCRIPTION
**Pull request by dgryski**

### CODE REVIEW

**Comment by dgryski** [comment]
Weird CI failure on Windows.

**Comment by Integralist** [comment]
@dgryski I'm going to re-run the Windows job because I honestly don't understand why that randomly failed.

File: pkg/config/data_test.go [hunk 1]
```diff
@@ -35,10 +35,11 @@ func TestConfigLoad(t *testing.T) {
...
 	)
 	if err := f.Load("foo", "/path/to/config.toml", c); err != nil {
-		if !errors.As(err, &fsterr.RemediationError{}) {
+		var remediation fsterr.RemediationError
+		if !errors.As(err, &remediation) {
 			t.Errorf("expected RemediationError got: %T", err)
 		}
-		if err.(fsterr.RemediationError).Remediation != fsterr.NetworkRemediation {
+		if remediation.Remediation != fsterr.NetworkRemediation {
 			t.Errorf("expected NetworkRemediation got: %s", err.(fsterr.RemediationError).Remediation)
```
Comment (Integralist) [inline pkg/config/data_test.go:43]: Should `err.(fsterr.RemediationError).Remediation` also change to just `remediation`?
```suggestion
			t.Errorf("expected NetworkRemediation got: %s", remediation.Remediation)
```

File: pkg/manifest/manifest_test.go [hunk 2]
```diff
@@ -109,7 +109,7 @@ func TestManifest(t *testing.T) {
 			} else {
 				// otherwise if we expect the manifest to be invalid/unrecognised then
 				// the error should match our expectations.
-				if !errors.As(err, &tc.expectedError) {
+				if !errors.Is(err, tc.expectedError) {
```
Comment (Integralist) [inline pkg/manifest/manifest_test.go:112]: I tried this change locally and see:

Screenshot 2022-08-03 at 09 14 13

Comment (Integralist) [reply pkg/manifest/manifest_test.go:112]: oh nevermind, just noticed that this is errors.Is not errors.As

**Comment by Integralist** [comment]
@dgryski no idea why this is suddenly failing on Windows.

File: pkg/config/data_test.go (hunk 1 above)
Comment (dgryski) [reply pkg/config/data_test.go:43]: Oops,  yes.

**Comment by dgryski** [comment]
Huh, this failed on my Mac (and I'm pretty sure it passed yesterday).  Only difference is I upgraded to 1.19
```
--- FAIL: TestApplication (0.09s)
    --- FAIL: TestApplication/help_argument_only (0.04s)
        run_test.go:58:   strings.Join({
              	... // 1095 identical bytes
              	"                  alphanumeric character and contain only\n      ",
              	"                           alphanumeric characters, underscores,",
            - 	" and\n",
            + 	"\n ",
              	"                                ",
            + 	"and",
```

I'll figure out what's going on here..

**Comment by dgryski** [comment]
The failure is more obvious under Mac.   It's due to this patch from the initial tinygo support PR:

```
diff --git a/pkg/text/text.go b/pkg/text/text.go
index 1e0d46d..9483cdf 100644
--- a/pkg/text/text.go
+++ b/pkg/text/text.go
@@ -163,25 +163,25 @@ func Break(w io.Writer) {
 // Error is a wrapper for fmt.Fprintf with a bold red "ERROR: " prefix.
 func Error(w io.Writer, format string, args ...interface{}) {
        format = strings.TrimRight(format, "\r\n") + "\n"
-       fmt.Fprintf(w, "\n"+BoldRed("ERROR: ")+format, args...)
+       fmt.Fprintf(w, "\n"+Wrap(BoldRed("ERROR: ")+format, DefaultTextWidth)+"\n", args...)
 }

 // Warning is a wrapper for fmt.Fprintf with a bold yellow "WARNING: " prefix.
 func Warning(w io.Writer, format string, args ...interface{}) {
        format = strings.TrimRight(format, "\r\n") + "\n"
-       fmt.Fprintf(w, "\n"+BoldYellow("WARNING: ")+format, args...)
+       fmt.Fprintf(w, "\n"+Wrap(BoldYellow("WARNING: ")+format, DefaultTextWidth)+"\n", args...)
 }

 // Info is a wrapper for fmt.Fprintf with a bold "INFO: " prefix.
 func Info(w io.Writer, format string, args ...interface{}) {
        format = strings.TrimRight(format, "\r\n") + "\n"
-       fmt.Fprintf(w, "\n"+Bold("INFO: ")+format, args...)
+       fmt.Fprintf(w, "\n"+Wrap(Bold("INFO: ")+format, DefaultTextWidth)+"\n", args...)
 }

 // Success is a wrapper for fmt.Fprintf with a bold green "SUCCESS: " prefix.
 func Success(w io.Writer, format string, args ...interface{}) {
        format = strings.TrimRight(format, "\r\n") + "\n"
-       fmt.Fprintf(w, "\n"+BoldGreen("SUCCESS: ")+format, args...)
+       fmt.Fprintf(w, "\n"+Wrap(BoldGreen("SUCCESS: ")+format, DefaultTextWidth)+"\n", args...)
 }
```

**Comment by dgryski** [comment]
Formatting difference due to this patch from the tinygo support PR:

```
diff --git a/pkg/text/text.go b/pkg/text/text.go
index 1e0d46d..9483cdf 100644
--- a/pkg/text/text.go
+++ b/pkg/text/text.go
@@ -163,25 +163,25 @@ func Break(w io.Writer) {
 // Error is a wrapper for fmt.Fprintf with a bold red "ERROR: " prefix.
 func Error(w io.Writer, format string, args ...interface{}) {
        format = strings.TrimRight(format, "\r\n") + "\n"
-       fmt.Fprintf(w, "\n"+BoldRed("ERROR: ")+format, args...)
+       fmt.Fprintf(w, "\n"+Wrap(BoldRed("ERROR: ")+format, DefaultTextWidth)+"\n", args...)
 }

 // Warning is a wrapper for fmt.Fprintf with a bold yellow "WARNING: " prefix.
 func Warning(w io.Writer, format string, args ...interface{}) {
        format = strings.TrimRight(format, "\r\n") + "\n"
-       fmt.Fprintf(w, "\n"+BoldYellow("WARNING: ")+format, args...)
+       fmt.Fprintf(w, "\n"+Wrap(BoldYellow("WARNING: ")+format, DefaultTextWidth)+"\n", args...)
 }

 // Info is a wrapper for fmt.Fprintf with a bold "INFO: " prefix.
 func Info(w io.Writer, format string, args ...interface{}) {
        format = strings.TrimRight(format, "\r\n") + "\n"
-       fmt.Fprintf(w, "\n"+Bold("INFO: ")+format, args...)
+       fmt.Fprintf(w, "\n"+Wrap(Bold("INFO: ")+format, DefaultTextWidth)+"\n", args...)
 }

 // Success is a wrapper for fmt.Fprintf with a bold green "SUCCESS: " prefix.
 func Success(w io.Writer, format string, args ...interface{}) {
        format = strings.TrimRight(format, "\r\n") + "\n"
-       fmt.Fprintf(w, "\n"+BoldGreen("SUCCESS: ")+format, args...)
+       fmt.Fprintf(w, "\n"+Wrap(BoldGreen("SUCCESS: ")+format, DefaultTextWidth)+"\n", args...)
 }
```

**Comment by Integralist** [comment]
Hmm. That change didn't cause an error in the PR that it was implemented so it's odd it's failing now.

**Comment by Integralist** [comment]
This is the PR that introduced that change (and we can see the CI passed):
https://github.com/fastly/cli/runs/7540097472?check_suite_focus=true

**Comment by Integralist** [comment]
OK after some testing I've found the problem gets introduced with go version 1.18.5 on my mac.

**Comment by Integralist** [comment]
I'm closing this PR in favour of: https://github.com/fastly/cli/pull/623 as I had to fix these same issues to fix the Windows issue. Thank you @dgryski for fixing these issues so I could just copy/paste them into my PR 🙇🏻

**Pull request closed without merging**